#!/usr/bin/env python3
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator

PACKAGE = "sr_piezo_feedback"

gen = ParameterGenerator()

gen.add("contact_time", double_t, 0, "Fading time after contact", 0.25 , 0.25,   2)

gen.add("max_amplitude", double_t, 0, "Maximum amplitude on output signal", 1.0, 0.5, 1.0)
gen.add("min_amplitude", double_t, 0, "Minimum amplitude on output signal", 0.2, 0, 0.4)
gen.add("max_frequency", double_t, 0, "Maximum frequency on output signal", 80.0, 30, 150.0)
gen.add("min_frequency", double_t, 0, "Minimum frequency on output signal", 1.0, 0.1, 10.0)

gen.add("pst_saturation", double_t, 0, "PST_SATURATION", 550.0, 500.0, 1200.0)

exit(gen.generate(PACKAGE, "sr_piezo_feedback_node", "SrPiezoFeedbackPST"))
